package.json

```diff
@@ -115,55 +115,49 @@
-    "proxy:start": "cd proxy && cargo build --bin git-remote-rad && cargo run",
-    "proxy:start:release": "cd proxy && cargo build --release --bin git-remote-rad && cargo run --release",
-    "proxy:start:test": "cd proxy && cargo build --bin git-remote-rad && cargo run -- --test",
-    "proxy:start:test-watch": "cd proxy && cargo build --bin git-remote-rad && cargo watch -x 'run -- --test'",
-    "cypress:run": "yarn run cypress run",
-    "cypress:open": "yarn run cypress open",
+    "dist": "yarn _private:dist:clean && yarn _private:rollup:build && yarn _private:proxy:build:release && electron-builder --publish never",
     "release": "scripts/release.js",
-    "release:finalize": "scripts/release.js --finalize",
-    "postinstall": "patch-package",
-    "prettier": "prettier \"**/*.@(js|ts|json|svelte|css|html)\" --ignore-path .gitignore",
-    "prettier:check": "yarn prettier --check",
-    "prettier:write": "yarn prettier --write",
+    "typescript:check": "tsc --noEmit && tsc --noEmit --project cypress && svelte-check",
+    "prettier:check": "yarn _private:prettier --check",
+    "prettier:write": "yarn _private:prettier --write",
     "lint": "eslint . --ignore-path .gitignore --ext .js,.svelte,.ts --max-warnings=0",
-    "reset:state": "scripts/reset-state.sh"
+    "reset:state": "scripts/reset-state.sh",
+    "manage:id": "scripts/manage-identity.sh",
+    "_private:start": "yarn _private:proxy:build:release && yarn run-p --race _private:rollup:watch _private:electron:start",
+    "_private:start:dev": "yarn _private:proxy:build && yarn run-p --race _private:rollup:watch _private:electron:start",
+    "_private:test:integration": "wait-on tcp:17246 && yarn _private:rollup:build && yarn run cypress run; status=$?; [ \"$CI\" = true ] && kill `pidof radicle-proxy`; exit $status",
+    "_private:test:integration:debug": "wait-on tcp:17246 && yarn run cypress open",
+    "_private:electron:start": "wait-on ./public/bundle.js ./native/main.comp.js && NODE_ENV=development electron .",
+    "_private:dist:clean": "rm -rf ./dist && mkdir ./dist && yarn _private:proxy:clean && yarn _private:rollup:clean",
+    "_private:rollup:clean": "rm -rf public/bundle.* && rm -f native/main.comp.js",
+    "_private:prettier": "prettier \"**/*.@(js|ts|json|svelte|css|html)\" --ignore-path .gitignore",
+    "_private:proxy:build": "cd proxy && cargo build --all-features --all-targets",
+    "_private:proxy:build:release": "cd proxy && cargo build --release",
+    "_private:proxy:clean": "cd proxy && cargo clean",
+    "_private:proxy:start:test": "cd proxy && cargo build --bin git-remote-rad && cargo run -- --test",
+    "_private:proxy:start:test:watch": "cd proxy && cargo build --bin git-remote-rad && cargo watch -x 'run -- --test'",
+    "_private:rollup:build": "yarn _private:rollup:clean && rollup -c",
+    "_private:rollup:watch": "yarn _private:rollup:clean && rollup -c -w",
+    "postinstall": "patch-package"
```

Comment:
Is this a normal amount of behaviour in the scripts section of a package.json. It seems rather peculiar that so much operational knowledge is captured in here. Are there plans to have a more sustainable way to manage these scripting needs?

Reply:
We could extract this into a separate script and call it from `package.json`. For that we would have to write some process management boilerplate, or use the `wait-on` and `npm-run-all` packages programmatically in that script. @geigerzaehler also pointed out that this could be a nice refactoring.

I didn't want to do it in this issue to keep the scope down. Should I open an issue about it?